custom_components/localtuya/pytuya/__init__.py

```diff
@@ -1125,6 +1149,14 @@ def _generate_payload(self, command, data=None, gwId=None, devId=None, uid=None)
                 json_data["uid"] = uid
             else:
                 json_data["uid"] = self.id
+        if "cid" in json_data:
+            if nodeId is not None:
+                json_data["cid"] = nodeId
+            elif self.node_id != "" and self.node_id is not None:
+                json_data["cid"] = self.node_id
+            else:
+                # don't send cid for non-subdevices
+                del json_data["cid"]
```

Comment:
Ah, potentially this is why you needed to copy() a lot.

Reply:
Yes. But I did it too much. only applies for the JSON objects. They get written to directly in the payload_dict and persist otherwise.